packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1006,4 +1012,86 @@ export class KubernetesClient {
       );
     }
   }
+
+  async createIngressesInformer(id?: number): Promise<number> {
```

Comment:
Instead of having to add a function for each class,

Should we instead having a map constant of values so it makes it easier for the calls?

Ex. If PODS is passed in, we map and check the table, get /apis/networking.k8s.io/v1/*/pods 

For ingresses, do the same, etc.





For example,

Instead of createIngressesInformer 

It's more generic / uses createInformer.

Reply:
Sorry i was busy on other things that had higher priority. Now i'm back to this.
The problem is that we also need to provide a ListPromise object to the informer which has type `() => Promise<{response, body}>`. Within this function we need to set the apiCLient and call the according listNamespaced function with the current namespace. 

Even though we could create a map that has a function that returns what we need as a value (we need to pass the current namespace), I wonder if it's worth it, just to try to save some couple of rows each time we lose readibility. I would skip this if there is not a strong will.